frontend/client/src/components/ResetPasswordModal.js

```diff
@@ -0,0 +1,161 @@
+import React, { createRef } from 'react'
+import Modal from '../components/Modal'
+import PendingOperationButton from '../components/PendingOperationButton'
+import { auth } from '../store/firebase'
+import { withStore } from '../store'
+import Logging from '../util/logging'
+import firebase from 'firebase'
+
+class ResetPasswordModalBase extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentPassword: '',
+      password: '',
+      confirmPassword: '',
+      passwordsMatch: true,
+      passwordIsValid: false,
+      successful: false,
+      message: '',
+      formHasBeenEdited: false,
+      confirmPasswordHasBeenEdited: false,
+    }
+    this.onChange = this.onChange.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+    this.canSubmit = this.canSubmit.bind(this)
+    this.handleFirebaseError = this.handleFirebaseError.bind(this)
+
+    this.toast = createRef()
+  }
+
+  onChange(event) {
+    let fieldName = event.target.name
+    let fieldContent = event.target.value
+    // Serialize updates
+    this.setState((state) => {
+      let newState = {
+        formHasBeenEdited: true,
+      }
+      if (fieldName === 'current-password') {
+        newState.currentPassword = fieldContent
+      }
+      if (fieldName === 'new-password') {
+        newState.password = fieldContent
+        newState.passwordIsValid = newState.password && newState.password.length >= 6
+        newState.passwordsMatch = newState.password === state.confirmPassword
+      }
+      if (fieldName === 'confirm-password') {
+        newState.confirmPasswordHasBeenEdited = true
+        newState.confirmPassword = fieldContent
+        newState.passwordsMatch = state.password === newState.confirmPassword
+      }
+      return newState
+    })
+  }
+
+  handleFirebaseError(err) {
+    Logging.error(err)
+    if (err.code === 'auth/wrong-password') {
+      this.props.onFailure('Current password incorrect, please try again')
+    } else if (err.code === 'auth/too-many-requests') {
+      this.props.onFailure('Too many incorrect attempts. Please try again later')
+    } else {
+      this.props.onFailure('Failed to reset password, please try again')
+    }
```

Comment:
I like this addition to the error handling method! We should add it to `ChangePasswordModal` as well in case of any unexpected errors there.